pkg/edgeview/src/system.go

```diff
@@ -1306,6 +1315,140 @@ func runTechSupport(cmds cmdOpt, isLocal bool) {
+	if err != nil {
+		fmt.Printf("runCollectInfo, can not create collect info request file\n")
+		return
+	}
+
+	err = monitorFileDeletion(requestCollectInfoFile, startTime, 20*time.Minute)
+	if err != nil {
+		log.Errorf("runCollectInfo: monitor file deletion error %v", err)
+		fmt.Println("runCollectInfo, monitor file deletion error")
+		return
+	}
+
+	targzFilename, err := findEveInfoFile(startTime)
+	if err != nil {
+		log.Errorf("runCollectInfo: can not find the eve info file, %v", err)
+		fmt.Printf("runCollectInfo, can not find the eve info file\n")
+		return
+	}
+
+	// Copy the info tar.gz file to the client side
+	err = runCopy("cp/"+targzFilename, nil)
+	if err != nil {
+		log.Errorf("runCollectInfo: copy kube file error %v", err)
+	}
+
+	// ask to remove this tar.gz file
+	err = touchRequestFile(requestRemoveInfoGZFile, targzFilename)
+	if err != nil {
+		log.Errorf("runCollectInfo: generate remove request file, %v", err)
+	}
+}
+
+func monitorFileDeletion(filePath string, startTime time.Time, timeout time.Duration) error {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.Errorf("monitorFileDeletion: %v", err)
+		return err
+	}
+	defer watcher.Close()
+
+	done := make(chan bool)
+	go func() {
+		for {
+			select {
+			case event, ok := <-watcher.Events:
+				if !ok {
+					return
+				}
+				if event.Op&fsnotify.Remove == fsnotify.Remove {
+					if event.Name == filePath {
+						log.Noticef("monitorFileDeletion: File deleted: %s", event.Name)
+						done <- true
+						return
+					}
+				}
+			case err, ok := <-watcher.Errors:
+				if !ok {
+					return
+				}
+				log.Noticef("monitorFileDeletion:: error: %v", err)
+			case <-time.After(timeout - time.Since(startTime)):
+				log.Noticef("monitorFileDeletion: Timeout exceeded")
+				done <- true
+				return
+			}
+		}
+	}()
+
+	dir := filepath.Dir(filePath)
+	if err := watcher.Add(dir); err != nil {
+		log.Errorf("monitorFileDeletion: %v", err)
+		return err
+	}
+
+	<-done
+	return nil
+}
+
+func touchRequestFile(filename, content string) error {
+	file, err := os.OpenFile(filename, os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0644)
+	if err != nil {
+		return err
+	}
+	defer file.Close() // Ensure the file is closed after it's opened/created.
+
+	if content != "" {
+		if _, err := file.WriteString(content); err != nil {
+			return err // Return the error if writing the content fails
+		}
+	}
+
+	return nil
+}
+
+func findEveInfoFile(createTime time.Time) (string, error) {
+	matches, err := filepath.Glob(infoFilePattern)
+	if err != nil {
+		return "", err // Return the error if any occurs during the file search
+	}
+
+	// Look for a the right file generated
+	for _, match := range matches {
+		fileInfo, err := os.Stat(match)
+		if err != nil {
+			continue // If we can't stat the file, skip it
+		}
+
+		// Check if the file was modified after the create time
+		if fileInfo.ModTime().After(createTime) {
+			return match, nil // Return the first recent file found
```

Comment:
What happens if there is more than one match?

Reply:
there should not be any file matches the pattern of collect-info tarball, and the timestamp is after we started collection.

Reply:
What if the node crashes and reboot during creation, then there can be two, can't there?

Reply:
The file is in /run, i.e. not persisted, so this should not happen. Also the filename is constant.
@naiming-zededa Correct?

Reply:
> The one before the reboot is older than this time.

How will it be cleaned up?

Reply:
no plan to clean is up, like if users manually run collect-info.sh and they may not delete this.

Reply:
one thing we can do is let the debug container to check, if a collect-info file in /persist has been around more than 2 weeks, it can go ahead to delete this file.

Reply:
implemented the task in debug-tasks.sh for removing this type of tarball files if older than 10 days

Reply:
Hmm, I am not sure if that is a good idea, because it also cleans up the collect-info output manually triggered, doesn't it?
Then:
1. 10 days are too short, perhaps a user created a collect-info and wants to copy it later to a usb stick
2. 10 days are too long, perhaps a user expects after 2 hours that the space has been freed again and they can use it for another vm

Reply:
for 1) we'll ask them edgeview can do for them now, no need to log in and trigger then wait for 20 days to port to usb anymore
for 2) it's better than we leave them now unnoticed. since using edgeview, this will not likely to happen as before, mainly is to clean up unexpected cases.

Reply:
But the user might have created the collect-info output not with edgeview or might have some created with edgeview and some on the console.

Reply:
yes, but we either decide to clean them up, if they forget about them, or we don't since we have no idea how long they want to leave them and come back to fetch them later. So, i'm taking a middle point, and think after 10 days, they probably forget about them. Most people would immediately download them but they may forget to delete the one on the device.

Reply:
I fear the following could happen:

User is somewhere out in the field without fast internet connection.
A problem happens.
They call their IT support staff and they tell them (over the phone) to run `collect-info.sh` and reboot.
After three weeks they come home and the IT support wants to extract the collect info output and it is gone.

(The IT support is too lazy to also explain how to copy that file to a usb stick)

Reply:
Ok, I'll remove this task then.

Reply:
@christoph-zededa I changed the task actually, so now it only deletes edgeview generated old tarball files, your IT support guy should be fine now :-)

Reply:
:-)